src/sinks/datadog/traces/apm_stats/integration_tests.rs

```diff
@@ -0,0 +1,425 @@
+/// The output of the sink is our HTTP server.
+/// Each member (topology, shutdown) of the Return value of this function must be kept
+/// in scope by the caller until the test is done.
+///
+/// The sink is run with a max batch size of one.
+/// The two trace events are intentionally configured within the same time bucket window.
+/// This creates a scenario where the stats payload that is output by the sink after processing the
+/// *second* batch of events (the second event) *should* contain the aggregated statistics of both
+/// of the trace events. i.e , the hit count for that bucket should be equal to "2" , not "1".
+async fn start_vector() -> (RunningTopology, tokio::sync::mpsc::UnboundedReceiver<()>) {
+    let dd_agent_address = format!("0.0.0.0:{}", vector_receive_port());
+
+    let source_config = toml::from_str::<DatadogAgentConfig>(&format!(
+        indoc! { r#"
+            address = "{}"
+            multiple_outputs = true
+        "#},
+        dd_agent_address,
+    ))
+    .unwrap();
+
+    let mut builder = ConfigBuilder::default();
+    builder.add_source("in", source_config);
+
+    let dd_traces_endpoint = format!("http://127.0.0.1:{}", server_port_for_vector());
+    let cfg = format!(
+        indoc! { r#"
+            default_api_key = "atoken"
+            endpoint = "{}"
+            batch.max_events = 1
+        "#},
+        dd_traces_endpoint
+    );
+
+    let api_key = std::env::var("TEST_DATADOG_API_KEY")
+        .expect("couldn't find the Datatog api key in environment variables");
+    assert!(!api_key.is_empty(), "TEST_DATADOG_API_KEY required");
+    let cfg = cfg.replace("atoken", &api_key);
+
+    let sink_config = toml::from_str::<DatadogTracesConfig>(&cfg).unwrap();
+
+    builder.add_sink("out", &["in.traces"], sink_config);
+
+    let config = builder.build().expect("building config should not fail");
+
+    let (topology, shutdown) = start_topology(config, false).await;
+    info!("Started vector.");
+
+    (topology, shutdown)
+}
+
+/// An end-to-end test which validates the APM stats payloads output by of the `datadog_traces` sink are
+/// correct by comparing them with the same APM stats payloads output by the Agent.
+/// Two Agent containers are initialized, and fed the same trace data.
+/// One Agent container feeds into an HTTP server where we parse the stats. The other Agent
+/// container feeds traces into the Datadog Agent source of Vector, which outputs to the traces
+/// sink and finally the same HTTP server to process stats payloads.
+/// The fields of the stats payloads are then compared.
+///
+/// This test specifically verifies the Aggregation of stats across multiple batches of events
+/// through vector. Consumers of the Agent's stats payloads expect the Aggregation of stats for a
+/// 20 second time window on any given stats payload within that time window.
+#[tokio::test]
+async fn apm_stats_e2e_test_dd_agent_to_vector_correctness() {
```

Comment:
I'm not seeing where the 20 seconds is specified in this test. Likely because I'm not understanding it properly. Any ideas where that comes from?

Reply:
This comment needed to be updated actually regardless of your question so I appreciate that.

I've updated the comment to be more specific about where the bucket window is defined within Vector, and also state where the requirement is coming from.
I don't think the APM stats backend of DD is open source so we unfortunately can't drop a link to that.
But the other comments within traces/apm_stats/ do point to the Agent side of the solution.